fix: display the correct receipt on a links from the donation history page #3778

mehul0810 opened the pull request:
## Description
This PR resolves #3778 

## How Has This Been Tested?
I've tested this PR as per the acceptance criteria in the issue. Please check the video for more information.

## Screenshots (jpeg or gifs if applicable):
**Video Link:** https://www.useloom.com/share/5be24b695a0342bc99efb68f4fc2e383

## Types of changes
Bug fix (a non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

ravinderk commented on includes/misc-functions.php:
```diff
@@ -847,58 +847,64 @@ function array_column( $input = null, $columnKey = null, $indexKey = null ) {
  * @return bool Whether the receipt is visible or not.
  */
-function give_can_view_receipt( $payment_key = '' ) {
+function give_can_view_receipt( $donation_id ) {
```
@mehul0810 Add backward compatibility for `payment_key` as we discussed

mehul0810 replied:
@ravinderk Added backward compatibility as discussed for the function `give_can_view_receipt`, in case if called directly.

mehul0810 commented:
@ravinderk I've tested and fixed minor issues I found after merging latest pull and implementing suggested changes. Also, added a new video for the missed scenario in earlier video.

## Video for "Confirm Email" Scenario.
https://www.useloom.com/share/0ce016f2e2c04d34913e8b722c2da754

ravinderk commented:
@mehul0810 

Email Access workflow is working fine for me on donation history page but not on the donation confirmation page:

![image](https://user-images.githubusercontent.com/1784821/48764934-0b821b80-ecd7-11e8-87b2-52a4cd29dfd5.png)

mehul0810 commented:
@ravinderk To resolve the issue you mentioned on donation confirmation is due to early usage of `get_receipt` AJAX placeholder. So, I've used it after finalizing all the conditional checks.

Check commit: https://github.com/impress-org/give/pull/3825/commits/f1d071217cc25f28473d3c43d9240422efc1185f
